tools/schema_loader.cc

```diff
@@ -662,10 +663,18 @@ future<std::vector<schema_ptr>> load_schemas(db::config& dbcfg, std::string_view
 future<schema_ptr> load_one_schema_from_file(db::config& dbcfg, std::filesystem::path path) {
     return async([&dbcfg, path] () mutable {
         auto schemas = do_load_schemas(dbcfg, read_file(path));
-        if (schemas.size() != 1) {
-            throw std::runtime_error(fmt::format("Schema file {} expected to contain exactly 1 schema, actually has {}", path.native(), schemas.size()));
+        if (schemas.size() == 1) {
+            return std::move(schemas.front());
+        } else if (schemas.size() == 2) {
+            // We expect a base table at index 0 and a view/index on it at index 1
+            if (!schemas[0]->is_view() && schemas[1]->is_view() && schemas[0]->id() == schemas[1]->view_info()->base_id()) {
+                return std::move(schemas[1]);
+            }
         }
-        return std::move(schemas.front());
+        throw std::runtime_error(fmt::format(
+                    "Schema file {} expected to contain exactly 1 schema or 2 schemas (base table and view), actually has {} non-related schemas",
+                    path.native(),
+                    schemas.size()));
```

Comment:
It would be better to load an arbitrary schema [1], and have a parameter indicate which table/index/view we're talking about. This way you could point the tool at a schema dump (from DESCRIBE SCHEMA) and use --table table-or-view-or-index-name to indicate the table. I believe DESCRIBE SCHEMA always names the index.

[1] It's common in databases to use "schema" to refer to all tables' schema, not just one, but we're stuck with this inconsistent terminology

Reply:
I agree and I do have plans to address this. It is annoying to have to delete all but the interesting table definition from a schema received from support (they usually dump the entire DESCRIBE into a single file).
Note, that we already allow looking up a specific schema, using `--keyspace` and `--table`, when loading the schema from system tables. The same should work for `schema.cql`.

I opened https://github.com/scylladb/scylladb/issues/16638 for this, because this PR is long enough already and this falls outside its scope.